### PR TITLE
feat(vdp): add component definition list endpoint

### PR DESCRIPTION
**Pull request by jvallesm**
Because

- Component page requires unified filtering and pagination.

This commit

- Adds a "component list" endpoint to the public VDP API.
- Unifies the component view and removes connector and operator view enums.
  - Connector and operator definition messages are unified in `component_definition.proto`
- Replaces `CONNECTOR_BLOCKCHAIN` by `CONNECTOR_APPLICATION` ([INS-3739](https://linear.app/instill-ai/issue/INS-3739/rename-blockhain-type-to-application))
- Removes operator / connector definition lookup endpoints (unused).

### CODE REVIEW

File: vdp/pipeline/v1beta/component_definition.proto [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+}
+
+// ConnectorType defines the connector type based on its task features.
+enum ConnectorType {
+  // Unspecified.
+  CONNECTOR_TYPE_UNSPECIFIED = 0;
+  // Source connector.
+  CONNECTOR_TYPE_SOURCE = 1;
+  // Destination connector.
+  CONNECTOR_TYPE_DESTINATION = 2;
```
Comment (jvallesm) [inline vdp/pipeline/v1beta/component_definition.proto:59]: @donch1989 can these 2 enum values be deprecated as well? I don't see them being used.

File: vdp/pipeline/v1beta/component_definition.proto [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+// GetConnectorDefinitionResponse contains the requested connector definition.
+message GetConnectorDefinitionResponse {
+  // The connector definition resource.
+  ConnectorDefinition connector_definition = 1;
+}
+
+// LookUpConnectorDefinitionAdminRequest represents a request by an admin to
+// query a connector definition by its UID.
+message LookUpConnectorDefinitionAdminRequest {
```
Comment (jvallesm) [inline vdp/pipeline/v1beta/component_definition.proto:284]: @donch1989 are the `LookUp[Connector|Operator]DefinitionAdmin` endpoints in use? Can we remove them?

File: vdp/pipeline/v1beta/component_definition.proto (hunk 1 above)
Comment (donch1989) [reply vdp/pipeline/v1beta/component_definition.proto:59]: Yes, they are deprecated, can remove them.

File: vdp/pipeline/v1beta/component_definition.proto (hunk 2 above)
Comment (donch1989) [reply vdp/pipeline/v1beta/component_definition.proto:284]: They are not in use as well. I think we can remove them.

Comment (jvallesm) [reply vdp/pipeline/v1beta/component_definition.proto]: Done 🧹

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**